android-instrumentation/src/main/java/co/elastic/apm/android/instrumentation/okhttp/client/OkHttpClientAdvice.java

```diff
@@ -34,6 +34,6 @@ public static void enter(@Advice.Argument(0) OkHttpClient.Builder builder) {
         OkHttpContextStore contextStore = new OkHttpContextStore();
         OtelOkHttpEventListener.Factory otelFactory = new OtelOkHttpEventListener.Factory(contextStore);
         builder.eventListenerFactory(new CompositeEventListenerFactory(otelFactory, builder.getEventListenerFactory$okhttp()));
-        builder.addInterceptor(new OtelOkHttpInterceptor(contextStore));
+        builder.interceptors().add(0, new OtelOkHttpInterceptor(contextStore));
```

Comment:
This is the only production code change, the rest is some testing setup needed to verify this change.

Reply:
I missed this when I wrote my comment. Now I understand why this PR contains seemingly unrelated changes.

Reply:
No worries, hopefully this setup will be needed only once, further tests will just use it so this kind of big changes should only happen once ;)